src/fklearn/validation/evaluators.py

```diff
@@ -603,6 +603,66 @@ def spearman_evaluator(test_data: pd.DataFrame,
     return {eval_name: score}
 
 
+@curry
+def ndcg_evaluator(test_data: pd.DataFrame,
+                   prediction_column: str = "prediction",
+                   target_column: str = "target",
+                   k: int = None,
+                   exponential_gain: bool = True,
+                   eval_name: str = None) -> EvalReturnType:
+    """
+    Computes the Normalized Discount Cumulative Gain (NDCG) between
+    of the original and predicted rankings:
+    https://en.wikipedia.org/wiki/Discounted_cumulative_gain
+
+    Parameters
+    ----------
+
+    test_data : Pandas DataFrame
+        A Pandas' DataFrame with with target and prediction scores.
+
+    prediction_column : String
+        The name of the column in `test_data` with the prediction scores.
+
+    target_column : String
+        The name of the column in `test_data` with the target.
+
+    k : int, optional (default=None)
+        The size of the rank that is used to fit (highest k scores) the NDCG score. If None, use all outputs.
+
+    exponential_gain : bool (default=True)
+        If False, then use the linear gain. The exponential gain places a stronger emphasis on retrieving
+        relevant items. If the relevance of these items is binary values in {0,1}, then the two approaches
+        are the same, which is the linear case.
+
+    Returns
+    ----------
+    log: dict
+        A log-like dictionary with the NDCG score.
+    """
+
+    if eval_name is None:
+        eval_name = f"ndcg_evaluator__" + target_column
```

Comment:
Removes `f` from the beginning of the string or use f-string interpolation, the second one is better.
```suggestion
        eval_name = f"ndcg_evaluator__{target_column}"
```

Reply:
Thanks!